tests/framework/installer/ceph_manifests.go

```diff
@@ -81,7 +85,18 @@ func (m *CephManifestsMaster) Settings() *TestCephSettings {
 	return m.settings
 }
 
-func (m *CephManifestsMaster) GetCRDs(k8shelper *utils.K8sHelper) string {
+func (m *CephManifestsMaster) GetCRDs(t *testing.T, k8shelper *utils.K8sHelper) string {
+	k8sVersion, err := k8sutil.GetK8SVersion(k8shelper.Clientset)
+	if err != nil {
+		assert.NoError(t, err)
+	}
+	minValidatinPolicy := k8sVersion.AtLeast(version.MustParseSemantic(MinValidatinPolicy))
+	// For multi-cluster integration we create a internal mode cluster and external mode cluster both
+	// but since CRDs are not namespaced tight we need to use pre-1.25/crds.yaml so that CI works.
+	if !minValidatinPolicy || m.settings.ClusterName == "test-external" || m.settings.ClusterName == "multi-cluster" {
```

Comment:
Isn't the crds.yaml version only dependent on the k8s version? I don't see why the external or multi-cluster tests would use the pre-1.25 crds. That would be an indication that external clusters are broken with the new CEL rules.
```suggestion
	if !minValidatinPolicy {
```

Reply:
did you take time to read my comments just above this check
```text
// For multi-cluster integration we create a internal mode cluster and external mode cluster both
// but since CRDs are not namespaced tight we need to use pre-1.25/crds.yaml so that CI works.
```

Reply:
I thought we did some work to update other-namespace `common-second-cluster.yaml` recently. What about those?

Reply:
Correct, the CRDs are not namespaced. But the version of the CRDs used (CEL vs non-CEL) only depends on what version of K8s, so I still don't understand why the multi-cluster test would need to use the non-CEL version of the CRDs.

Reply:
Let's discuss in huddle, but the issue happening, in once cluster where CRD with CEL rule is created. We are creating two cephCluster in different namespace, one is internal mode where we define `mons`, `osd`  and so and in second cluster we create external cluster, then the cel rule through the error because the cephcluster crd external mode can't have `mon` and `osd` field. So,

Reply:
This means the CEL rule is not ever allowing an external cluster at the same time as a converged cluster. Sounds like a bug in the CEL. We need to allow that.

Reply:
What was the error in the CI you were seeing? I was able to create both a converged cecphcluster CR and an external CephCluster CR in my local minikube cluster without hitting any errors from the CEL. And I don't see any issues in the external cluster yaml [created by the test](https://github.com/rook/rook/blob/580760c8d38a921620cd06836669724b59fec615/tests/framework/installer/ceph_manifests.go#L616-L630), so I would expect it to work.

Reply:
I removed this check and see the multiclustertest is failing and when I check the logs I can see 
```yaml
2023-11-29 05:50:24.396462 E | ceph-cluster-controller: failed to reconcile CephCluster "multi-external/test-external". failed to add finalizer: failed to add finalizer "cephcluster.ceph.rook.io" on "test-external": CephCluster.ceph.rook.io "test-external" is invalid: [spec: Invalid value: "object": external clusters cannot define the mon settings, spec: Invalid value: "object": external cluster cannot define dashboard settings, spec: Invalid value: "object": external cluster cannot define monitoring settings, spec: Invalid value: "object": external cluster cannot define network settings, spec: Invalid value: "object": external cluster cannot define storage settings, spec: Invalid value: "object": external cluster cannot define disruptionManagement settings, spec: Invalid value: "object": external cluster cannot define mgr settings]
```

Reply:
I was also able to create the external crd locally

Reply:
Actually I was able to see that error locally in the operator log, I had only been looking for errors when running the `kubectl create` command. When the operator tries to add the finalizer, some empty fields are causing the problem with the CEL of the external cluster. So this is really a problem for any external cluster, and not just a problem with the CI in that scenario with the multi-cluster test.

After I remove the CEL from the CRDs and allow the CR to be updated, this is what the spec looks like for the external CephCluster CR. So the CEL rules need to allow for the empty structs. And if it's not simple to get working, let's remove those CEL rules for the external cluster so we can get this merged, then we can follow up with a separate PR.
```yaml
  spec:
    cephVersion: {}
    cleanupPolicy:
      sanitizeDisks: {}
    crashCollector:
      disable: true
    dashboard: {}
    disruptionManagement: {}
    external:
      enable: true
    healthCheck:
      daemonHealth:
        mon:
          interval: 45s
        osd: {}
        status: {}
    logCollector: {}
    mgr: {}
    mon: {}
    monitoring: {}
    network:
      multiClusterService: {}
    security:
      keyRotation:
        enabled: false
      kms: {}
    storage:
      flappingRestartIntervalHours: 0
      store: {}
```
And let's make the change from the original comment:
```
if !minValidatinPolicy {
```